editor/plugins/path_3d_editor_plugin.cpp

```diff
@@ -260,60 +301,126 @@ void Path3DGizmo::redraw() {
 		}
 
 		const Transform3D *r = frames.ptr();
-		Vector<Vector3> v3p;
-		for (int i = 0; i < sample_count - 1; i++) {
+		Vector<Vector3> _collision_segments;
+		Vector<Vector3> bones;
+		Vector<Vector3> ribbon;
+		for (int i = 0; i < sample_count; i++) {
 			const Vector3 p1 = r[i].origin;
-			const Vector3 p2 = r[i + 1].origin;
 			const Vector3 side = r[i].basis.get_column(0);
 			const Vector3 up = r[i].basis.get_column(1);
 			const Vector3 forward = r[i].basis.get_column(2);
 
-			// Curve segment.
-			v3p.push_back(p1);
-			v3p.push_back(p2);
+			// Collision segments.
+			if (i != sample_count) {
+				const Vector3 p2 = r[i + 1].origin;
+				_collision_segments.push_back(p1);
+				_collision_segments.push_back(p2);
+			}
+
+			// Path3D as a ribbon.
+			ribbon.push_back(p1);
 
 			// Fish Bone.
-			v3p.push_back(p1);
-			v3p.push_back(p1 + (side + forward + up * 0.3) * 0.06);
+			const Vector3 p_left = p1 + (side + forward - up * 0.3) * 0.06;
+			const Vector3 p_right = p1 + (-side + forward - up * 0.3) * 0.06;
+			bones.push_back(p1);
+			bones.push_back(p_left);
 
-			v3p.push_back(p1);
-			v3p.push_back(p1 + (-side + forward + up * 0.3) * 0.06);
+			bones.push_back(p1);
+			bones.push_back(p_right);
 		}
 
-		add_lines(v3p, path_material);
-		add_collision_segments(v3p);
+		add_collision_segments(_collision_segments);
+		add_lines(bones, path_material);
+		add_vertices(ribbon, path_material, Mesh::PRIMITIVE_LINE_STRIP);
 	}
 
-	// 2. Draw handles.
+	// 2. Draw handles when selected.
 	if (Path3DEditorPlugin::singleton->get_edited_path() == path) {
-		Vector<Vector3> v3p;
-		Vector<Vector3> handle_points;
-		Vector<Vector3> sec_handle_points;
-
-		for (int i = 0; i < c->get_point_count(); i++) {
-			Vector3 p = c->get_point_position(i);
-			handle_points.push_back(p);
-			// Push out points first so they get selected if the In and Out points are on top of each other.
-			if (i < c->get_point_count() - 1) {
-				v3p.push_back(p);
-				v3p.push_back(p + c->get_point_out(i));
-				sec_handle_points.push_back(p + c->get_point_out(i));
+		PackedVector3Array handle_lines;
+		PackedVector3Array primary_handle_points;
+		PackedVector3Array secondary_handle_points;
+		PackedInt32Array _secondary_handle_ids; // Avoid shadowing member on Node3DEditorGizmo
```

Comment:
```suggestion
		PackedInt32Array _secondary_handle_ids; // Avoid shadowing member on Node3DEditorGizmo.
```
Though I'd suggest renaming it instead to something that doesn't shadow but isn't just differentiated by an underscore, easy to make mistakes and get confused by that